### PR TITLE
Various fixes to better support targeting in command handlers

### PR DESCRIPTION
**Pull request by cdupuis**

### CODE REVIEW

**Comment by ddgenome** [comment]
@cdupuis is this ready for review?

**Comment by cdupuis** [comment]
@ddgenome feel free to review. There isn't much there.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,13 @@ All notable changes to this project will be documented in this file.
...
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
-## [Unreleased](https://github.com/atomist/automation-client-ts/compare/1.1.0...HEAD)
+## [Unreleased](https://github.com/atomist/automation-client-ts/compare/1.2.0...HEAD)
+
+### Added
+
+-   Provide a dummy id to InMemoryProject. [#438](https://github.com/atomist/automation-client/issues/438)
+
+## [1.2.0](https://github.com/atomist/automation-client-ts/compare/1.1.0...1.2.0) - 2018-12-27
```
Comment (ddgenome) [inline CHANGELOG.md]: Why would this be here?

File: lib/project/mem/InMemoryProject.ts [hunk 2]
```diff
@@ -38,7 +38,11 @@ export class InMemoryProject extends AbstractProject {
      * Create a new InMemoryProject without an id
      */
     public static of(...files: Array<{ path: string, content: string } | File>): InMemoryProject {
-        return InMemoryProject.from(undefined, ...files);
+        return InMemoryProject.from({
+            owner: "dummyOwner",
+            repo: "dummyRepo",
+            url: "https://fakeGitHub.com/dummyOwner/dummyRepo",
+        }, ...files);
```
Comment (ddgenome) [inline lib/project/mem/InMemoryProject.ts]: Wasn't this from another commit?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@atomist/automation-client",
-  "version": "1.2.0",
+  "version": "1.2.1",
```
Comment (ddgenome) [inline package.json]: ? Maybe this branch needs rebased?

**Pull request merged**